output/changelog.go

```diff
@@ -0,0 +1,315 @@
+package output
+
+import (
+	"fmt"
+
+	"github.com/tufin/oasdiff/checker"
+	"github.com/tufin/oasdiff/checker/localizations"
+	"github.com/tufin/oasdiff/diff"
+	"github.com/tufin/oasdiff/report"
+)
+
+type changelogProfile struct {
+	diffReport        *diff.Diff
+	operationsSources *diff.OperationsSourcesMap
+	checks            checker.BackwardCompatibilityCheckConfig
+}
+
+func (p *changelogProfile) format(format string, diffReport *diff.Diff, operationsSources *diff.OperationsSourcesMap) (diffOutput []byte, errCode int, err error) {
```

Comment:
the idea here is to decompose the diffReport into individual changes and run all checkers against these changes.
If no checker matches the decomposed diffReport, then the code will fallback to the text output (and over time add more checkers).

Reply:
Not sure that I understand what is meant by "decompose" but I prefer the output format to be deterministic so that a user always gets what s/he specified. This is important specially for users who programmatically process the output.